akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -703,6 +703,25 @@ final case class Referer(uri: Uri) extends jm.headers.Referer with RequestHeader
   def getUri: akka.http.javadsl.model.Uri = uri.asJava
 }
 
+//https://tools.ietf.org/html/rfc7231#section-7.1.3
+object `Retry-After` extends ModeledCompanion[`Retry-After`] {
+  def apply(delaySeconds: Long): `Retry-After` = apply(Left(delaySeconds))
+  def apply(timestamp: DateTime): `Retry-After` = apply(Right(timestamp))
+}
+final case class `Retry-After`(delaySecondsOrDateTime: Either[Long, DateTime]) extends jm.headers.RetryAfter with ResponseHeader {
+  require(delaySecondsOrDateTime.left.toOption.forall(_ > 0), "Retry-after header must not contain a negative delay in seconds")
```

Comment:
Can delaySeconds be zero?
The check can be reduced to:

```scala
delaySecondsOrDateTime.left.forall(...)
```

Reply:
nice, thanks

yes, I checked again and I made a mistake here with the validation check

will fix soon